.travis.yml

```diff
@@ -0,0 +1,57 @@
+language: php
+
+sudo: required
+
+services:
+  - docker
+
+addons:
+  apt:
+    packages:
+      - docker-ce
+
+cache:
+  directories:
+    - $HOME/.composer/cache/files
+
+notifications:
+  email: false
+
+env:
+  global:
+    - DOCKER_COMPOSE_VERSION=1.18.0
+    - THREADS=4
+  matrix:
+    # Devportal module tests.
+    - PHP_VERSION=7.1 DEPENDENCIES="--prefer-lowest" TEST_ROOT="modules/api_reference/tests"
+    - PHP_VERSION=7.1 DEPENDENCIES="--prefer-lowest" TEST_ROOT="modules/tests"
+matrix:
+  allow_failures:
+    - env: PHP_VERSION=7.1 DRUPAL_CORE=8.7.x-dev DEPENDENCIES="--prefer-lowest" TEST_ROOT="modules/api_reference/tests"
+    - env: PHP_VERSION=7.1 DRUPAL_CORE=8.7.x-dev DEPENDENCIES="--prefer-lowest" TEST_ROOT="modules/tests"
+  fast_finish: true
+
+# TODO Cache Docker images and PHP service build (common parts of the built, like Drupal core install, for a day).
+before_install:
+  - sudo rm /usr/local/bin/docker-compose
+  - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+  - chmod +x docker-compose
+  - sudo mv docker-compose /usr/local/bin
+  - cd .travis && chmod u+x *.sh
+  - docker-compose up -d --build
+  - docker-compose ps
+
+script:
+  - set -e
+  # Do not report warnings (like Drupal.Semantics.FunctionT.NotLiteralString) on Travis CI. Use selective disabling
+  # when Drupal Coder will support PHPCS >= 3.2.0.
+  # https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file
+  - docker-compose run phpcs -s -n --colors --standard=/opt/drupal-module/phpcs.xml.dist /opt/drupal-module
+  # Do not exist if a PHPUnit test fails, we would like to run everything from
+  # after_* sections.
+  - set +e
+  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then docker-compose run php /opt/drupal-module/.travis/run-test.sh; fi
```

Comment:
The if is not needed here. In the original repo, tests can't be run in with pull requests for security reasons, but here this is not the case.